### PR TITLE
Fix Klarna  redirect flow for declined payments

### PR DESCRIPTION
**Pull request by Onein**
If a customer completes the Klarna payment flow and the payment is
declined, Klarna assumes that the customer can choose another payment
method offered by Klarna e.g. credit card. The plugin  is currently limited
to pay later and results in order creation even though the payment failed.
To prevent this a check is now in place which cancles the order creation
in that case, so the customer can choose an alternative.

Signed-off-by: Tim Marx <tim.marx2@gmail.com>

### CODE REVIEW

File: Snippets/frontend/plugins/payment/stripe_payment/base.ini [hunk 1]
```diff
@@ -5,6 +5,7 @@ payment_error/message/invalid_owner_name = "The name provided in the billing add
 payment_error/message/redirect/failed = "Failed to prepare the payment flow."
 payment_error/message/redirect/internal_error = "An internal error occurred."
 payment_error/message/redirect/source_not_chargeable = "The payment processed was cancelled."
+payment_error/message/redirect/source_declined = "The payment source was declined, please choose a different payment method."
```
Comment (svenmuennich) [inline Snippets/frontend/plugins/payment/stripe_payment/base.ini:8]:
```suggestion
payment_error/message/redirect/source_declined = "The payment source was declined. Please choose a different payment method."
```

File: Snippets/frontend/plugins/payment/stripe_payment/base.ini [hunk 2]
```diff
@@ -17,6 +18,7 @@ payment_error/message/invalid_owner_name = "Der in der Rechnungsadresse angegebe
 payment_error/message/redirect/failed = "Der Bezahlvorgang konnte nicht vorbereited werden."
 payment_error/message/redirect/internal_error = "Ein interner Fehler ist aufgetreten."
 payment_error/message/redirect/source_not_chargeable = "Der Bezahlvorgang wurde abgebrochen."
+payment_error/message/redirect/source_declined = "Das Zahlungsmittel wurde abgelehnt, bitte wählen Sie eine andere Zahlungsart."
```
Comment (svenmuennich) [inline Snippets/frontend/plugins/payment/stripe_payment/base.ini:21]:
```suggestion
payment_error/message/redirect/source_declined = "Das Zahlungsmittel wurde abgelehnt. Bitte wählen Sie eine andere Zahlungsart."
```

**Review by svenmuennich — COMMENTED** [review]
Thanks for the PR. Looking good from my end.

@felixbrucker could you please take a look and give this a test?

**Comment by svenmuennich** [comment]
@Onein I just fixed the CI so that it is also run for PRs from forks. Could you please pull the latest changes?

**Review by felixbrucker — APPROVED** [review]
LGTM

**Comment by Onein** [comment]
Anything else needed before this can be merged?

**Review by svenmuennich — APPROVED** [review]

**Comment by svenmuennich** [comment]
Thank you @Onein for the contribution 🎉

**Pull request merged**